### PR TITLE
feat: add `OnyxAvatar` and `OnyxAvatarStack` component

### PR DESCRIPTION
**Pull request by larsrickert**
Relates to #947

Add OnyxAvatar component.

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/styles/mixins/sizes.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+* Mixin to define the given CSS variable for every onyx icon size.
+*/
+@mixin define-icon-sizes() {
```
Comment (BoppLi) [inline packages/sit-onyx/src/styles/mixins/sizes.scss:4]: we could name it more generic like define-rem-sizes or so? the implementation is not restricted to usage for icons at least ¯\\_(ツ)\_/¯

File: packages/sit-onyx/src/components/OnyxAvatarStack/OnyxAvatarStack.stories.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  }),
+};
+
+export default meta;
+type Story = StoryObj<typeof OnyxAvatarStack>;
+
+/**
+ * This example shows a default avatar stack.
+ */
+export const Default = {
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxAvatarStack/OnyxAvatarStack.stories.ts:28]: technically, you can actually simulate the "+5" features, the initials avatar works with "+x" 😜
image

(just not >9 ^^)

File: packages/sit-onyx/src/components/OnyxAvatar/OnyxAvatar.vue [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      width: 100%;
+      border-radius: inherit;
+
+      display: flex;
+      align-items: center;
+      justify-content: center;
+    }
+
+    @include sizes.define-icon-sizes using ($name, $size) {
+      @if $name != 12px {
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxAvatar/OnyxAvatar.vue:64]: why 12px? what am I missing? ^^

File: packages/sit-onyx/src/components/OnyxAvatar/OnyxAvatar.ct.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  });
+
+  // ASSERT
+  await expect(component).toContainText("AB");
+
+  // ACT
+  await component.update({ props: { label: "abcd" } });
+
+  // ASSERT
+  await expect(component).toContainText("AB");
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxAvatar/OnyxAvatar.ct.tsx:33]:
```suggestion
  // ACT
  await component.update({ props: { label: "abcd" } });

  // ASSERT
  await expect(component).toContainText("AB");
  
  // ACT
  await component.update({ props: { label: "a" } });

  // ASSERT
  await expect(component).toContainText("A");
```

**Review by BoppLi — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxAvatarStack/OnyxAvatarStack.stories.ts (hunk 2 above)
Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxAvatarStack/OnyxAvatarStack.stories.ts:28]: True :D I will ask Jannick tomorrow for a "real" implementation, I can imagine that it has another background color then

File: packages/sit-onyx/src/components/OnyxAvatar/OnyxAvatar.vue (hunk 3 above)
Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxAvatar/OnyxAvatar.vue:64]: I assumed that 12px is too small for reading the initials so I excluded 12px for now.
 I will double check this with Jannick tomorrow

File: packages/sit-onyx/playwright/snapshots/components/OnyxAvatarStack/wrap-chromium-linux.png
Comment (BoppLi) [inline packages/sit-onyx/playwright/snapshots/components/OnyxAvatarStack/wrap-chromium-linux.png:1]: will this feature be reflected in figma?

**Review by BoppLi — APPROVED** [review]

File: packages/sit-onyx/playwright/snapshots/components/OnyxAvatarStack/wrap-chromium-linux.png
Comment (larsrickert) [reply packages/sit-onyx/playwright/snapshots/components/OnyxAvatarStack/wrap-chromium-linux.png:1]: I will check with Jannick

**Review by BoppLi — DISMISSED** [review]

**Review by BoppLi — APPROVED** [review]

**Pull request merged**